src/features/session_replay/aggregate/index.js

```diff
@@ -152,46 +155,67 @@ export class Aggregate extends AggregateBase {
         raw: true
       }, this)
 
+      registerHandler('recordReplay', () => {
+        // if it has aborted or BCS returned bad entitlements, do not allow
+        if (this.blocked || !this.entitled) return
+        // if it isnt already (fully) initialized... initialize it
+        if (!recorder) this.initializeRecording(false, true, true)
+        // its been initialized and imported the recorder but its not recording (mode === off || error)
+        else if (this.mode !== MODE.FULL) this.switchToFull()
+        // if it gets all the way to here, that means a full session is already recording... do nothing
+      }, this.featureName, this.ee)
+
+      registerHandler('pauseReplay', () => {
+        this.actuallyStop(this.mode !== MODE.ERROR)
```

Comment:
Shouldn't we force harvest if the mode is error? Confusing.

Reply:
if the mode is error and we force a harvest -- it will take the 30s buffer we've been storing up and harvest it -- which we should only actually want to get a data harvest if running in full mode